packages/web-components/fast-components/src/accordion/accordion.ts

```diff
@@ -0,0 +1,139 @@
+import { attr, booleanConverter, FASTElement, observable } from "@microsoft/fast-element";
+import {
+    keyCodeArrowDown,
+    keyCodeArrowUp,
+    keyCodeEnd,
+    keyCodeHome,
+    wrapInBounds,
+} from "@microsoft/fast-web-utilities";
+import { StartEnd } from "../patterns/start-end";
+import { applyMixins } from "../utilities";
+import { FASTAccordionItem } from "./accordion-item";
+
+export enum AccordionExpandMode {
+    single = "single",
+    multi = "multi",
+}
+
+export class Accordion extends FASTElement {
+    @attr({ attribute: "expand-mode" })
+    public expandmode: AccordionExpandMode = AccordionExpandMode.multi;
+
+    @observable
+    public accordionItems: HTMLElement[];
+    public accordionItemsChanged(oldValue, newValue): void {
+        if (this.$fastController.isConnected) {
+            this.removeItemListeners(oldValue);
+            this.accordionIds = this.getItemIds();
+            this.setItems();
+        }
+    }
+
+    private activeid: string;
+    private activeItemIndex: number = 0;
+    private accordionIds: Array<string | null>;
+
+    private change = (): void => {
+        this.$emit("change");
+    };
+
+    private setItems = (): void => {
+        console.log(this.isSingleExpandMode());
+        this.accordionIds = this.getItemIds();
+        this.accordionItems.forEach((item: HTMLElement, index: number) => {
+            if (item instanceof FASTAccordionItem) {
+                item.addEventListener("change", this.activeItemChange);
+                if (this.isSingleExpandMode()) {
+                    this.activeItemIndex !== index
+                        ? (item.expanded = false)
+                        : (item.expanded = true);
+                }
+            }
+            const itemId: string | null = this.accordionIds[index];
+            item.setAttribute(
+                "id",
+                typeof itemId !== "string" ? `accordion-${index + 1}` : itemId
```

Comment:
Can't "`accordion-${index + 1}`" result in duplicate id's if you have multiple accordions?

Reply:
They are in their own shadow DOMs. Duplicate ids across different shadow DOMs is acceptable.

Reply:
Really? did not not know that.  So Query selector by id on the document won't find elements within web components then?